### PR TITLE
sqlcapture: validate existance and accessibility of table

### PR DESCRIPTION
**Pull request by mdibaiee**
**Description:**

- At `Validate` and `ApplyUpsert` RPC, make sure we have access to the binding tables
- I'm reusing our Discover logic for finding accessible tables here to be consistent. The code path that I have is probably not the most efficient if user is not choosing all the discovered bindings, but it is _simple_ and means we don't need some new logic.

**Workflow steps:**

- Create a capture to a sql endpoint (e.g. `source-postgres`) and add a binding to a nonexistant (or non-accessible) table. You should see an error like this:
```
    could not find or access table estuary_test_cfngjj
```

**Documentation links affected:**

(list any [documentation links](https://docs.google.com/document/d/1SRC9VS9zyCzWl3n4HXHbc4wPB1eLxJHkA2rtu9ZNokM/edit?usp=sharing) that you created, or existing ones that you've identified as needing updates, along with a brief description)

**Notes for reviewers:**

(anything that might help someone review this PR)

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/connectors/523)

### CODE REVIEW

File: sqlcapture/main.go [hunk 1]
```diff
@@ -79,16 +79,35 @@ func (d *Driver) Validate(ctx context.Context, req *pc.ValidateRequest) (*pc.Val
...
 	var out []*pc.ValidateResponse_Binding
 	for _, binding := range req.Bindings {
 		var res Resource
 		if err := pf.UnmarshalStrict(binding.ResourceSpecJson, &res); err != nil {
 			return nil, fmt.Errorf("error parsing resource config: %w", err)
 		}
+
+		var streamID = JoinStreamID(res.Namespace, res.Stream)
+
+		if !SliceContains(streamID, discoveredStreams) {
```
Comment (willdonnelly) [inline sqlcapture/main.go:141]: Unless I've missed some subtlety the whole `discoveredStreams` / `SliceContains` thing could be removed. The `discoveredBindings` map is already keyed by stream ID, so `if _, ok := discoveredBindings[streamID]; ok {` should be equivalent here.

Comment (mdibaiee) [reply sqlcapture/main.go:141]: @willdonnelly I checked the return value of `DiscoverCatalog` and it's `[]*pc.DiscoverResponse_Binding` so it's not a map, but an array

Comment (willdonnelly) [reply sqlcapture/main.go:141]: Ah, my mistake. I was thinking of the underlying `DiscoverTables` operation that each database's interface code implements, which returns the map I was thinking of (and then `DiscoverCatalog` implements the whole "translate that into a slice of `pc.DiscoverResponse_Binding` objects with appropriate JSON schemas" step).

Comment (mdibaiee) [reply sqlcapture/main.go]: Thanks, I switched to using that one now 👍🏽

**Pull request merged**